src/lex.cpp

```diff
@@ -743,27 +776,52 @@ void lexer::parse_binary_number() {
   this->input_ = check_garbage_in_number_literal(input);
 }
 
-char8* lexer::check_garbage_in_number_literal(char8* input) {
-  char8* garbage_begin = input;
+void lexer::parse_octal_number(bool strict, bool is_0o) {
+  if (is_0o) assert(strict);
+  char8* input = this->input_;
+
+  while (this->is_octal_digit(*input)) {
+    ++input;
+  }
+
+  if (!is_0o) {
+    switch (*input) {
+    QLJS_CASE_DECIMAL_DIGIT:
```

Comment:
Could we use `is_decimal_digit` here?

Reply:
Would this now be `parse_decimal_digits_and_underscores`?